test/e2e/common/secrets.go

```diff
@@ -48,6 +48,27 @@ var _ = framework.KubeDescribe("Secrets", func() {
 		doSecretE2EWithMapping(f, &mode)
 	})
 
+	It("should be able to mount in a volume regardless of a different secret existing with same name in different namespace", func() {
+		var (
+			namespace2  *api.Namespace
+			err         error
+			secret2Name = "secret-test-" + string(uuid.NewUUID())
+		)
+
+		if namespace2, err = f.CreateNamespace("secret-namespace", nil); err != nil {
+			framework.Failf("unable to create new namespace %s: %v", namespace2.Name, err)
+		}
+
+		secret2 := secretForTest(namespace2.Name, secret2Name)
+		secret2.Data = map[string][]byte{
+			"this_should_not_match_content_of_other_secret": []byte("similarly_this_should_not_match_content_of_other_secret\n"),
+		}
+		if secret2, err = f.ClientSet.Core().Secrets(namespace2.Name).Create(secret2); err != nil {
+			framework.Failf("unable to create test secret %s: %v", secret2.Name, err)
+		}
+		doSecretE2EWithoutMapping(f, nil, secret2.Name)
```

Comment:
nit: When it is unclear what the value being passed is for, put the parameter name as a comment for clarity. e.g. `doSecretE2EWithoutMapping(f, nil /* defaultMode */, secret2.Name)`